Updated install, dev and contributing docs (#4378)
sunpy/sunpy: CONTRIBUTING.rst

Each issue should have a series of labels that provide information about the nature of the issue.
If you find an issue you'd like to work on, please make sure to add a comment to let people know that you are working on it! This will make it less likely that effort is duplicated.

.. note::

>     sunpy is Free and open-source software (FOSS), under the BSD-2 license. By contributing you are stating that you have the right to and agree to have your work distributed under the terms of this license.

Author: Good point.

>     sunpy is Free and open-source software (FOSS), under the BSD-2 license. By contributing you are stating that you have the right to and agree to have your work distributed under the terms of this license.

Author: Fixed.

Each issue should have a series of labels that provide information about the nature of the issue.
If you find an issue you'd like to work on, please make sure to add a comment to let people know that you are working on it! This will make it less likely that effort is duplicated.

.. note::

>     sunpy is Free and open-source software (FOSS), under the BSD-2 license. By contributing you are stating that you have the right to and agree to, have your work distributed under the terms of this license.

Reviewer: ```suggestion
    sunpy is Free and open-source software (FOSS), under the BSD-2 license. By contributing you are stating that you have the right to and agree to have your work distributed under the terms of this license.
```

>     This applies to everyone who wants to contribute on work time no matter who their employer is.

Reviewer: ```suggestion
    This applies to everyone who wants to contribute during work time no matter who their employer is.
```

>     If you work at a university or institution, you will need to check if you can contribute code you have written at work.

Reviewer 2: This should probably be slightly expanded to include everyone on work time no matter who ever their employer is, although obviously it makes sense for us to focus on unis.

Reviewer 2: Also we might want to consider adding a link to this or something in the PR template?

Author: Added, see what you think.

    You should start by checking if there is a Open Source Software Policy (e.g., `Standford's policy <https://otl.stanford.edu/open-source-stanford>`__) for your work place.

>     If not, `OSS-Watch <http://oss-watch.ac.uk/resources/contributing>`__ summaries what you will need to check and who to ask, however this more aimed at a United Kingdom readers.

Reviewer: ```suggestion
    If not, `OSS-Watch <http://oss-watch.ac.uk/resources/contributing>`__ summaries what you will need to check and who to ask, however this resource is aimed at a UK readers.
```

>     `Standford's guidance <https://otl.stanford.edu/sites/g/files/sbiybj10286/f/otlcopyrightguide.pdf>`__ for example allows someone to contribute and open source their code.

Reviewer: ```suggestion
   As an example, `Standford's guidance <https://otl.stanford.edu/sites/g/files/sbiybj10286/f/otlcopyrightguide.pdf>`__ allows someone to contribute and open source their code.
```

>    As an example, `Standford's guidance <https://otl.stanford.edu/sites/g/files/sbiybj10286/f/otlcopyrightguide.pdf>`__ allows someone to contribute and open source their code.

Reviewer 2: `../CONTRIBUTING.rst:94: WARNING: Block quote ends without a blank line; unexpected unindent.`

Author: The issue here looks like one bullet point was not indented right.
